pipeline_steps/common.groovy

```diff
@@ -1625,35 +1625,44 @@ void createComponentGateTrigger(String name, String repoUrl, String jiraProjectK
   }
 }
 
+List getMatchingJobs(String regexMatch){
+  List jobNames = Jenkins.getInstance().getAllItems()*.fullName
+  List matchingJobs = []
+  // Cannot do for (job in jobNames), see:
```

Comment:
I'm pretty sure that's not the case here because there is no closure. With the for-loop for the builds `parallelBuilds` is a map of `String` to `Closure` whereas here is simply a `List` of `String`.

Reply:
The script console is a good place for trying stuff like this out. Also @NonCPS methods can be used for accessing more interesting groovy language functions.

Reply:
Yep, it seems to work:

```
List getMatchingJobs(String regexMatch){
  List jobNames = Jenkins.getInstance().getAllItems()*.fullName
  List matchingJobs = []
  for (job in jobNames) {
    if (job =~ /${regexMatch}/) {
      matchingJobs+=job
    }
  }
  return matchingJobs
}

println getMatchingJobs("^Setup-\\w+\$")
```